src/services/download-service.ts

```diff
@@ -12,123 +13,153 @@ class DownloadService {
     return !(this.player.isLive() || this.player.getVideoElement().mediaKeys);
   }
 
-  private isContentTypeSupported(response: Response) {
-    if (response.ok) {
-      const contentType = response.headers.get('content-type');
-      return !!(contentType && contentType.toLowerCase().includes('video'));
-    }
-    return false;
-  }
-
-  private getFilename(response: Response) {
-    if (this.player.isImage()) {
-      return this.player.sources?.metadata?.name || 'image';
-    }
-    const responseUrlSplit = response.url.split('/');
+  private getFilename(metadata: DownloadMetadata) {
+    if (this.player.sources.metadata?.name) return this.player.sources.metadata?.name;
+    if (this.player.isImage()) return 'image';
```

Comment:
the definition by product is to fall back to "image" if there's no name ?

Reply:
actually they haven't defined it.. it's just the file name when downloading and saving the file. basically the fallback will happen only if the image name is "" (empty) or for some reason the metadata in undefined but it really shouldn't happen.. i added the fallback just to be on the safe side.